src/swaps/astroport/AstroportSwapProvider.js

```diff
@@ -0,0 +1,286 @@
+        this.toTokenAddress
+      )
+    }
+
+    await this.sendLedgerNotification(quote.fromAccountId, 'Signing required to complete the swap.')
+
+    const swapTx = await client.chain.sendTransaction(txData)
+
+    const updates = {
+      status: 'WAITING_FOR_SWAP_CONFIRMATIONS',
+      swapTx,
+      swapTxHash: swapTx.hash
+    }
+
+    return {
+      id: uuidv4(),
+      fee: quote.fee,
+      slippage: 50,
+      ...updates
+    }
+  }
+
+  // ======== STATE TRANSITIONS ========
+
+  async waitForSwapConfirmations({ swap, network, walletId }) {
+    const client = this.getClient(network, walletId, swap.from, swap.fromAccountId)
+
+    try {
+      const tx = await client.chain.getTransactionByHash(swap.swapTxHash)
+      if (tx && tx.confirmations > 0) {
+        const { status } = await client.getMethod('getTransactionByHash')(swap.swapTxHash)
+        this.updateBalances({ network, walletId, assets: [swap.from] })
+        return {
+          endTime: Date.now(),
+          status
+        }
+      }
+    } catch (e) {
+      if (e.name === 'TxNotFoundError') console.warn(e)
+      else throw e
+    }
+  }
+
+  async performNextSwapAction(store, { network, walletId, swap }) {
+    let updates
+
+    if (swap.status === 'WAITING_FOR_SWAP_CONFIRMATIONS') {
+      updates = await withInterval(async () =>
+        this.waitForSwapConfirmations({ swap, network, walletId })
+      )
+    }
+
+    return updates
+  }
+
+  // ========= FEES ========
+
+  async estimateFees({ asset, txType, quote, feePrices }) {
+    if (txType !== AstroportSwapProvider.fromTxType) throw new Error(`Invalid tx type ${txType}`)
+
+    const nativeAsset = chains[cryptoassets[asset].chain].nativeAsset
+
+    const gasLimit =
+      quote.from === 'UST' || (quote.from === 'LUNA' && quote.to === 'UST') ? 400_000 : 1_500_000
+
+    const fees = {}
+
+    for (const feePrice of feePrices) {
+      let fee = BN(gasLimit).times(feePrice)
+      fees[feePrice] = unitToCurrency(cryptoassets[nativeAsset], fee).toFixed()
+    }
+
+    return fees
+  }
+
+  // ======== UTILS ========
+
+  _getRPC() {
+    const { chainID, nodeUrl } = TerraNetworks.terra_mainnet
+
+    return new LCDClient({
+      chainID,
+      URL: nodeUrl
+    })
+  }
+
+  _getDenom(asset) {
+    return {
+      LUNA: 'uluna',
+      UST: 'uusd'
+    }[asset]
+  }
+
+  async _getSwapRate(fromAmount, fromInfo, toInfo) {
+    const rpc = this._getRPC()
+
+    // Check coin types
+    const nativeToNative = fromInfo.type === 'native' && toInfo.type === 'native'
+    const erc20ToErc20 = fromInfo.type === 'erc20' && toInfo.type === 'erc20'
+    const nativeToErc20 = fromInfo.type === 'native' && toInfo.type === 'erc20'
+    const erc20ToNative = fromInfo.type === 'erc20' && toInfo.type === 'native'
+
+    // Select correct query and address depending on coin types
+    let contractData = {
+      address: '',
+      query: ''
+    }
+
+    if (nativeToNative) {
+      this.fromTokenAddress = ''
```

Comment:
Amm I'm changing a prop of `this` which in this case is fromTokenAddress. So i need to have some global variables for the class because i receive some important information when `getQuote` method is executed which i need to use it when i do swap so i created these variables to store this data. It has been tested with all possible scenarios aka:
Native to native, ERC20 to ERC20, Native to ERC20 and ERC20 to Native and it works fine so far but if you have any idea how to store some global state in the class share with me. Probably i can create some global variables for the file outside the class scope and mutate them and use them in the class?